flake.nix

```diff
@@ -1,12 +1,40 @@
 {
   description = "Serokell Vault Tooling";
 
-  outputs = { self }: {
+  inputs.nixpkgs.url = "github:serokell/nixpkgs";
+  inputs.nix-unstable.url = "github:nixos/nix";
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+  inputs.flake-compat.url = "github:edolstra/flake-compat";
+  inputs.flake-compat.flake = false;
 
-    overlay = final: prev: {
-      vault-push-approles =
-        final.callPackage ./scripts/vault-push-approles.nix { };
-    };
+  outputs = { self, nixpkgs, flake-utils, ... }@inputs:
+    {
 
-  };
+      overlay = final: prev: {
+        vault-push-approles =
+          final.callPackage ./scripts/vault-push-approles.nix { };
+        vault-push-approle-envs =
+          final.callPackage ./scripts/vault-push-approle-envs.nix { };
+      };
+      nixosModules.vault-secrets = import ./modules/vault-secrets.nix;
+    } // (flake-utils.lib.eachSystem [ "x86_64-linux" ] (system:
```

Comment:
Looping over a single item list :100:

Reply:
We will likely want to add `aarch64-linux` here soon, so might as well do this

Reply:
I guess we better get an aarch64 machine we can use as remote builder then.

Reply:
Albali could do this with `binfmt.emulatedSystems = ["x86_64-linux"];`